In [ ]:
print('Hello World')

Hello World


# Basic Arithemtic Operations
`+ - * ** / // %`

In [ ]:
print(5+2) # Add
print(5-2) # Sub
print(5*2) # Multiplication
print(5/2) # Division

7
3
10
2.5


In [ ]:
print(5**2) # Power
print(5//2) #
print(5%2) # Mod

25
2
1


In [ ]:
1+2*3

7

In [ ]:
(1+2)*3

9

The order of operations, or operator precendence, for arithmetic expressions is as follows:

1. Expressions between parentheses
2. Exponents
3. Multiplication and division (from left to right)
4. Addition and subtraction (from left to right)

$ 2−1+8÷2^2⋅(1+1) $

$ 2−1+8÷2^2⋅(1+1)$  
$= 2−1+8÷2^2⋅2 $  
$= 2−1+8÷4⋅2$  
$= 2−1+2⋅2$  
$= 2−1+4$  
$= 1+4 = 5$


In [ ]:
2-1+8/2**2*(1+1)

5.0

## Exercise
$5^3\left(\frac{3}{5}+\frac{9}{2^3}\right)$

In [ ]:
5**3 * (3/5 + 9/2**3)

215.625

# Trigonometric function

In [ ]:
import math

In [ ]:
math.pi

3.141592653589793

In [ ]:
math.e

2.718281828459045

In [ ]:
math.sin(math.pi/3)

0.8660254037844386

|math|pthon|
|---|---|
|sine|math.sin()|
|cosine|math.cos()|
|tangent|math.tan()|
|||
|arcsine|math.asin()
|arccosine|math.acos()|
|arctangent|math.atan()|
|||
|hyperboli sine|math.sinh()|
|hyperbolic cosine|math.cosh()|
|hyperbolic tangent|math.tanh()|
|||
|hyperboli arcsine|math.asinh()|
|hyperbolic arccosine|math.acosh()|
|hyperbolic arctangent|math.atanh()|



In [ ]:
math.sinh(2)


3.626860407847019

## Example
1. $sin(30)$
2. $cos(45)$
3. $tan(60)$


# Exponental and logrithmic function
$e^3$  
$ln2$  
$log_{10}2$  
$\sqrt{4}$


In [ ]:
math.exp(3)

20.085536923187668

In [ ]:
math.e**3

20.085536923187664

In [ ]:
math.pow(math.e,3)

20.085536923187664

In [ ]:
math.log(10)

2.302585092994046

In [ ]:
math.log10(10)

1.0

In [ ]:
math.sqrt(4)

2.0

$\sqrt[3]{8}$

In [ ]:
8**(1/3)

2.0

In [ ]:
math.pow(8,1/3)

2.0

# Tip

In [ ]:
dir(math)

['__doc__',
 '__loader__',
 '__name__',
 '__package__',
 '__spec__',
 'acos',
 'acosh',
 'asin',
 'asinh',
 'atan',
 'atan2',
 'atanh',
 'ceil',
 'comb',
 'copysign',
 'cos',
 'cosh',
 'degrees',
 'dist',
 'e',
 'erf',
 'erfc',
 'exp',
 'expm1',
 'fabs',
 'factorial',
 'floor',
 'fmod',
 'frexp',
 'fsum',
 'gamma',
 'gcd',
 'hypot',
 'inf',
 'isclose',
 'isfinite',
 'isinf',
 'isnan',
 'isqrt',
 'lcm',
 'ldexp',
 'lgamma',
 'log',
 'log10',
 'log1p',
 'log2',
 'modf',
 'nan',
 'nextafter',
 'perm',
 'pi',
 'pow',
 'prod',
 'radians',
 'remainder',
 'sin',
 'sinh',
 'sqrt',
 'tan',
 'tanh',
 'tau',
 'trunc',
 'ulp']

In [ ]:
help(math.log)

Help on built-in function log in module math:

log(...)
    log(x, [base=math.e])
    Return the logarithm of x to the given base.
    
    If the base not specified, returns the natural logarithm (base e) of x.



# Exercise
(a) $4\frac{3}{5}+5\frac{9}{8}$  
(b) $43\left( \frac{\sqrt{513}}{6}+8^2 \right)+\left(\frac{54}{7}+5 .23^3 \right)^{0.75}$  
(c) $\frac{ln150}{3.2}+
7.5^{\frac{2}{3}}-
\left( \frac{\sqrt{135}+\sqrt[3]{250}} {e^{-0.3}} \right)^{\frac{1}{3}}$  
(d) $ \sin^2(\frac{4\pi}{3})\cos(\frac{5\pi}{6})+ \frac{\tan(\frac{\pi}{5} \ln 7)}{3+2\sqrt{3}}$

# Variable
- A variable name must start with a letter or the underscore character.
- A variable name cannot start with a number.
- A variable name can only contain alpha-numeric characters and underscores (A-z, 0-9, and _ )
- Variable names are case-sensitive (age, Age and AGE are three different variables)

In [3]:
a = 1; b=2; c= 3
print(a,b,c)
print(a+b)
print(b**c)

1 2 3
3
8


$$
\text{area} = \sqrt{s(s-a)(s-b)(s-c)} \> \text{where}\> s = \frac{a+b+c}{2}
$$
if `a=5, b=4, c=3` find area of the triangle?

In [4]:
import math
a = 5
b = 4
c = 3
s = (a+b+c)/2
area = math.sqrt(s*(s-a)*(s-b)*(s-c))
print(area)

6.0


## Exercise
if `x=8.67` find the solution.  
(a) $5x^3+2x^2-5x-7.5$  
(b) $\frac{e^{\sqrt{x}}+\log_{10}(x+2)}{\sqrt[3]{0.02+3x^3}}$  
(c) $\frac{\sqrt{\sin x + \cos^2 x}}{3\log_{10}(x-2)}$  
(d) $\frac{e^{1.58} - \sin 3x}{5} + \log_{10}(x+x^3)$  
(e) $ \arctan(2x+3) - \cot(x\pi)$

## Exercise 3
`a = -5.73, b= 17.5`
$$c = 2a+b/a$$  
$$d=4ac+3b$$  
(a) $c-\frac{d-a}{2c}+\frac{(a+c)^{2.5}}{\sqrt{abc}}$  
(b) $\log |(a-c)(b-a)|+\frac{a-b+c-d}{a+b-c+d}$  
(c) $a^2b - d^2c+(\frac{a}{c})^2-(\frac{b}{d})^{0.5}$